app/src/main/java/com/amaze/filemanager/filesystem/ftpserver/AndroidFtpFileSystemView.kt

```diff
@@ -98,6 +99,10 @@ class AndroidFtpFileSystemView(private var context: Context, root: String) : Fil
         // context = null!!
     }
 
+    private fun normalizePath(path: String): String {
+        return URI(path.replace(" ", "%20")).normalize().toString().replace("%20", " ")
```

Comment:
This is not a permanent fix. %20 is just for a space. Other special characters must also being url encoded. They should be decoded as well. Can't the `Uri.decode` be used?

Reply:
It is indeed hacky, and converting the space character to `%20` is only for `URI` to recognize it correctly.

Another alternative is to use [`FilenameUtils` from commons-io](http://commons.apache.org/proper/commons-io/javadocs/api-1.4/org/apache/commons/io/FilenameUtils.html).